modules/network/src/main/java/org/apache/ignite/internal/network/serialization/DeclaredType.java

```diff
@@ -29,11 +29,10 @@ public interface DeclaredType {
     int typeDescriptorId();
 
     /**
-     * Returns {@code true} if the type slot can only host (at runtime) instances of its declared type (and not subtypes),
-     * so the runtime marshalling type is known upfront. This is also true for enums, even though technically their values might
+     * Returns {@code true} if the contents of the type slot can only hove (at runtime) instances serialized as its declared type
```

Comment:
What's a "type slot"?

Reply:
It's either a field or a component of an array. So it's a slot for which some 'declared type' is statically known.

Reply:
Is this terminology used anywhere else? It is quite confusing...

Reply:
I don't know, I had to invent something myself. Any ideas are welcome :)